nvfuser/__init__.py

```diff
@@ -56,9 +56,41 @@ def __init__(self, id=None, max_length=1024):
         self.profiled = False
 
     def segment(self, inputs):
+        """
+        Decompose this FusionDefinition into a sequence of segment
+        FusionDefinitions.
+
+        This function runs the nvfuser segmentation algorithm and translates the
+        segments into their corresponding FusionDefinitions.
+
+        Args:
+            inputs (List[Union[Tensor, Scalar]]): A list of inputs to fusion.
+
+        Returns:
+            List[FusionDefinition]: The FusionDefinitions corresponding to the
+            sub-fusion segments of this FusionDefinition.
+        """
         num_segments = self._setup_segmentation(inputs)
+        if num_segments == 1:
+            self._finalize_segmentation()
+            return []
+
+        self.segments = []
+        self.segment_maps = []
+        self.last_used_segment = {}
```

Comment:
Please add a comment on what this dictionary is storing. The naming is not descriptive of the contents to me.

Reply:
Added ``` # Track the last segment a value is used as an input ```